tests/chacha20poly1305.cc

```diff
@@ -83,6 +83,61 @@ print_test(test_encrypt aead_encrypt,
   return ok;
 }
 
+// -----------------------------------------------------------------------------
+
+TEST(ApiSuite, ApiTest)
+{
+  // Documentation.
+  // Lines after START and before END are used in documentation.
+  {
+    // START OneShot
+    // Note: This is only an example, and you must bring your own random.
+
+    // Create a key ...
+    uint8_t key[32];
+    generate_random(key, 32);
+
+    // ... and a nonce.
+    uint8_t nonce[12];
+    generate_random(nonce, 12);
+
+    // We don't authenticate additional data in this example.
+    const char* aad = "";
+    const uint32_t aad_len = strlen(aad);
+
+    // This is our message.
+    const char* msg = "Hello, World!";
+    const uint32_t msg_len = strlen(msg);
+
+    // We need to allocate the same amount of memory for the ciphertext as for the plaintext ...
+    uint8_t* cipher = (uint8_t*)malloc(msg_len);
```

Comment:
This could as well be `msg` (in place). You could mention this here, but it certainly should go into the API description.
Same for decryption.